examples/fourier_transform_playground.py

```diff
@@ -14,18 +14,30 @@
 
 import numpy as np
 from magicgui import magic_factory
+from qtpy.QtWidgets import QApplication
 from scipy.fft import fft2, fftshift
 
 import napari
 from napari.qt.threading import thread_worker
 
 IMAGE_SIZE = 100
 FPS = 20
+FINISHED = False
 
 # meshgrid used to calculate the 2D sine waves
 x = np.arange(IMAGE_SIZE) - IMAGE_SIZE / 2
 X, Y = np.meshgrid(x, x)
 
+def wait_on_layers():
+    """Wait for the thread to finish.
+    To not close program before the thread finishes.
+    """
+    while "wave 0" not in viewer.layers:
+        sleep(0.1)
+        QApplication.processEvents()
+
+    QApplication.processEvents()
```

Comment:
This looks like it could be a more general utility! For now we can leave it in the example but in general it could be something in napari.qt?

```suggestion
def wait_for_layers(viewer, layer_names, *, interval=0.1):
    """Wait for any thread creating layers to finish.
    
    If we expect a different thread to be adding layers to
    `viewer`, we can wait for them to be added based on
    their expected names.
    """
    for name in layer_names:
        while name not in viewer.layers:
            sleep(interval)
            QApplication.processEvents()

    QApplication.processEvents()
```